src/mono/IosAppBuilderTasks/IosAppBuilderTask.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class IosAppBuilderTask : Task
+{
+    /// <summary>
+    /// Path to arm64 AOT cross-compiler (mono-aot-cross)
+    /// It's not used for x64 (Simulator)
+    /// </summary>
+    public string? CrossCompiler { get; set; }
+
+    /// <summary>
+    /// ProjectName is used as an app name, bundleId and xcode project name
+    /// </summary>
+    [Required]
+    public string ProjectName { get; set; } = ""!;
+
+    /// <summary>
+    /// Target directory with *dll and other content to be AOT'd and/or bundled
+    /// </summary>
+    [Required]
+    public string AppDir { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to Mono public headers (*.h)
+    /// </summary>
+    [Required]
+    public string MonoInclude { get; set; } = ""!;
+
+    /// <summary>
+    /// This library will be used as an entry-point (e.g. TestRunner.dll)
+    /// </summary>
+    [Required]
+    public string EntryPointLib { get; set; } = ""!;
```

Comment:
This needs a better name or description

Reply:
Um, I think it's pretty clear can you suggest a better name?

Reply:
Is this file name with file extension? What about `MainLibraryFileName`